lang/go/go.talon

```diff
@@ -66,25 +113,46 @@ else <user.text> [over]:
     key("enter")
     insert(user.formatted_text(text, "PRIVATE_CAMEL_CASE"))
 
-state while: "while "
+[state] if (err|error):
+    insert("if err != nil {")
+    key("enter")
+
+[state] if not (err|error):
+    insert("if err == nil {")
+    key("enter")
+
 while <user.text> [over]:
-    insert("while ")
-    insert(user.formatted_text(text, "PRIVATE_CAMEL_CASE"))
+    user.insert_between("for "," {}")
 
 state for: "for "
-for <user.text> [over]:
+state for <user.text> [over]:
     insert("for ")
     insert(user.formatted_text(text, "PRIVATE_CAMEL_CASE"))
 
+snip for range: "forr"
+
 state for range: "forr "
 range <user.text> [over]:
     insert("forr ")
     insert(user.formatted_text(text, "PRIVATE_CAMEL_CASE"))
 
+[state] for i:
+    user.insert_between("for i := 0; i< ", "; i++ {")
+
 state format: "fmt"
-format <user.text> [over]:
+state format <user.text> [over]:
     insert("fmt.")
     insert(user.formatted_text(text, "PUBLIC_CAMEL_CASE"))
+    insert("(\"")
+
+format print line [over]:
+    insert("fmt.Println")
+    insert("(\"")
+
+format print format [over]:
+    insert("fmt.Printf")
+    insert("(\"")
```

Comment:
These should just be in "funk" list (common functions).  In particular the more common one should just be "funk print", as that is standard in many languages